lib/round.rb

```diff
@@ -0,0 +1,111 @@
+
+class Round 
+
+    attr_reader :deck, :turns, :num_correct
+    def initialize(deck)
+        @deck = deck
+        @turns = []
+        @num_correct = []
+ 
+end
+
+def current_card
+    @deck.cards[@turns.count]
+end
+
+def take_turn(guess)
+    new_turn = Turn.new(guess, current_card)
+    @turns << new_turn
+    if new_turn.correct?
+        @num_correct << new_turn
+    end
+    new_turn
+end
+
+def number_correct
+ @num_correct.count    
+end
+
+
+def number_correct_by_category(category)
+    num_correct_in_category = []
+@turns.each do |turn|
+    if turn.correct? && turn.card.category == category
+        num_correct_in_category << turn
+    end
+end
+
+num_correct_in_category.count
```

Comment:
Similarly here, you could have your accumulator on line 31 be set to 0 instead of [], then you don't ned to count up the elements on line 38.

Reply:
It just seems like you're storing more than you need to in these variables. Objects in an array take up more space than a single integer. And if you don't need those objects, and rather the count, you might as well start with an integer.